client/lxd_server.go

```diff
@@ -141,3 +143,41 @@ func (r *ProtocolLXD) UseTarget(name string) InstanceServer {
 func (r *ProtocolLXD) IsAgent() bool {
 	return r.server != nil && r.server.Environment.Server == "lxd-agent"
 }
+
+// GetMetrics returns the text OpenMetrics data.
+func (r *ProtocolLXD) GetMetrics() (string, error) {
```

Comment:
Can this be a more structured type? Like map[string]any ?

Reply:
No, it's not a map, it's a text document with headers and a pretty complex structure.
We could in theory send it through the prometheus parser and return internal go-prometheus structs instead but that feels pretty wrong.

In practice, it's an OpenMetrics Text document so returning it as a string seems appropriate.

Reply:
Ok fair enough.